x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_rules_client.ts

```diff
@@ -114,7 +114,9 @@ export class RuleMigrationsDataRulesClient extends RuleMigrationsDataBaseClient
               {
                 doc: {
                   ...rest,
-                  elastic_rule: elasticRule,
+                  elastic_rule: elasticRule?.query
```

Comment:
Why do we need to do this here? 🤔

Reply:
I just realized that when we have untranslated rule the `query_language` won't be set and if I update query later we need to update `query_language` as well. Since we know that it always will be `esql`, I was thinking that we can always update `query_language` when we update the query.

Reply:
I would not include this logic in the data client layer, I think this condition should be handled by the application layer. Could we move that to the UI and make sure we always have a value for `query_language` when we update the `query`?

Reply:
Agree! Will move it to the UI